jujugui/static/gui/src/app/app.js

```diff
@@ -1933,6 +1933,16 @@ YUI.add('juju-gui', function(Y) {
       if (this.get('sandbox')) {
         console.log('switching models is not supported in sandbox');
       }
+      // Optionally update to the new base URL.
+      if (this.get('gisf')) {
+        var state = this.state;
+        var baseUrl = state.get('baseUrl');
+        var unsavedModelUrl = 'sandbox';
+        // Make sure we're not already in an unsaved model.
+        if (baseUrl.indexOf(unsavedModelUrl) < 0) {
```

Comment:
You can tell if you're in a real model by seeing if you're connected to one, have all of the credentials, etc.